File: exercise/java/day09/src/main/java/gift/GiftRequest.java
package gift;

public class GiftRequest {
    private final String giftName;
    private final boolean isFeasible;
    private final Priority priority;

    public GiftRequest(String giftName, boolean isFeasible, Priority priority) {
        this.giftName = giftName;
        this.isFeasible = isFeasible;
        this.priority = priority;
    }

    public String getGiftName() {
        return giftName;
    }

    public boolean isFeasible() {
        return isFeasible;
    }

    public Priority getPriority() {
        return priority;
    }
}

File: solution/day09/src/giftRequest.ts
export class GiftRequest {
    public readonly giftName: string;
    public readonly isFeasible: boolean;
    public readonly priority: Priority;

    constructor(giftName: string, isFeasible: boolean, priority: Priority) {
        this.giftName = giftName;
        this.isFeasible = isFeasible;
        this.priority = priority;
    }
}